model-optimizer/mo/front_ng/extractor.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging as log
+import re
+from collections import defaultdict
+from copy import copy
+
+import numpy as np
+
+from mo.utils.error import Error
+
+
+def fe_decodeNameWithPort(inputModel, node_name: str):
+    """
+    Decode name with optional port specification w/o traversing all the nodes in the graph
+    :param inputModel: Input Model
+    :param node_name:
+    :return: decoded place in the graph
+    """
+    # Check exact match with one of the names in the graph first
+    node = inputModel.get_place_by_tensor_name(node_name)
+    if node:
+        return node
+    # TODO: not tested for available frontends
+    regexpPost = r'(.*)(:(\d+))'
```

Comment:
nit: What does "Post" mean? Is it "Postfix"? If so, then let's use the full name

Reply:
Added TODO, delimiters and format for input node will be discussed and implemented separately